pkg/api/customization/multiclusterapp/validator_multiclusterapp.go

```diff
@@ -168,5 +172,23 @@ func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data
 	if err = ma.EnsureRoleInTargets(targetProjects, rolesToAdd, callerID); err != nil {
 		return err
 	}
+	// at this point we know the roles have changed, and that the user has the new roles in targets,
+	// retry underlying app upgrade
+	for _, t := range mcapp.Spec.Targets {
+		_, projectNS := ref.Parse(t.ProjectName)
+		if t.AppName != "" {
+			app, err := w.AppLister.Get(projectNS, t.AppName)
+			if err != nil || app == nil {
+				return fmt.Errorf("error %v getting app %s in %s", err, t.AppName, projectNS)
+			}
+			if val, ok := app.Labels["mcapp"]; !ok || val != mcapp.Name {
+				return fmt.Errorf("app %s in %s missing multi cluster app label", t.AppName, projectNS)
+			}
+			pv3.AppConditionUserTriggeredAction.True(app)
+			if _, err := w.Apps.Update(app); err != nil {
```

Comment:
do you think it would make sense to handle conflict error and do a retry a couple of times in a loop? As it's possible for the app to get updated on the backend, and 403 being hit. And we want to minimize user's doing api update when this error is hit